### DISCUSSION TITLE
How can I prevent major releases for conventional commits ?

### DISCUSSION THREAD

**Discussion by Bassel17** [nrwl/nx#28882] [Questions] [answered]
Am trying to setup nx releases in our mono-repo, and configure conventional commits and how they will behave based on the commits, it's pretty straight forward to change `minor` or `patch` of a release but how do we prevent `BREAKING CHANGE` or `!` from creating a major release. is there a straight forward way or do I have to use the nx/js programmatic api and create a release script and filter out those commits ?

**Comment by JamesHenry** [comment, marked as answer]
@Bassel17 by the definition of the spec you would not be following conventional commits if you did what you described: https://www.conventionalcommits.org/en/v1.0.0/ so it's not something that will be supported out of the box.

As you said, Nx release intentionally ships with a first-class programmatic API for folks who want to do custom things, so that is what I would recommend if you are certain you want to go this route.
